shared/secrets.js

```diff
@@ -0,0 +1,72 @@
+module.exports = {
+  // converts an object from serverless framework ({"a": "b", "c": "d"})
+  // to an array of secrets expected by the API :
+  // [{"key": "a", "value": "b"}, {"key": "c", "value": "d"}]
+  convertObjectToModelSecretsArray(obj) {
+    if (obj === {} || obj === null || obj === undefined) {
+      return [];
+    }
+    return Object.keys(obj)
+      .map(k => ({
+        key: k,
+        value: obj[k],
+      }));
+  },
+
+  // resolves a value from a secret
+  // if this is a raw value, return the value
+  // if this is a reference to a local environment variable, return the value of that env var
+  resolveSecretValue(key, value) {
+    const envVarRe = /^\${([^}]*)}$/;
+    const found = value.match(envVarRe);
+
+    if (!found) {
+      return value;
+    }
+
+    if (process.env[found[1]] === null || process.env[found[1]] === undefined) {
+      console.warn(`Env var ${found[1]} used in secret ${key} does not exist: this secret will not be created.`);
```

Comment:
why you don't use [this.serverless.cli.log](https://www.serverless.com/framework/docs/guides/plugins/cli-output) ?

Reply:
See [b09efa2](https://github.com/scaleway/serverless-scaleway-functions/pull/64/commits/b09efa2e89f8e4f75117b3be631a96fde0976905)